### PR TITLE
V1 search fields imported to V2

### PR DESCRIPTION
**Pull request by tareq89**
## Description

V1 all advanced search fields imported to V2

(https://github.com/opencrvs/opencrvs-core/issues/9435)
core pr: https://github.com/opencrvs/opencrvs-core/pull/9586
Farajaland PR: https://github.com/opencrvs/opencrvs-farajaland/pull/1396

## Checklist

- [x] I have linked the correct Github issue under "Development"
- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: src/form/v2/birth/advancedSearch.ts [hunk 1]
```diff
@@ -9,23 +9,196 @@
...
  */
 
-import { AdvancedSearchConfig, field } from '@opencrvs/toolkit/events'
+import { AdvancedSearchConfig, event, field } from '@opencrvs/toolkit/events'
+import format from 'date-fns/format'
+import subDays from 'date-fns/subDays'
 
+const statusOptions = [
+  {
+    value: 'ALL',
```
Comment (makelicious) [inline src/form/v2/birth/advancedSearch.ts:18]: Could we have enum for these values

File: src/form/v2/birth/advancedSearch.ts [hunk 2]
```diff
@@ -9,23 +9,196 @@
...
+]
+
+const timePeriodOptions = [
+  {
+    label: {
+      defaultMessage: 'Last 7 days',
+      description: 'Label for option of time period select: last 7 days',
+      id: 'form.section.label.timePeriodLast7Days'
+    },
+    value: `${format(subDays(new Date(), 7), 'yyyy-MM-dd')},${format(new Date(), 'yyyy-MM-dd')}`
```
Comment (makelicious) [inline src/form/v2/birth/advancedSearch.ts:106]: Could we have constants for values that are reused or magical

File: src/form/v2/birth/advancedSearch.ts [hunk 3]
```diff
@@ -9,23 +9,196 @@
...
+    },
+    value: `${format(subDays(new Date(), 90), 'yyyy-MM-dd')},${format(new Date(), 'yyyy-MM-dd')}`
+  },
+  {
+    label: {
+      defaultMessage: 'Last year',
+      description: 'Label for option of time period select: last year',
+      id: 'form.section.label.timePeriodLastYear'
+    },
+    value: `${format(subDays(new Date(), 365), 'yyyy-MM-dd')},${format(new Date(), 'yyyy-MM-dd')}`
```
Comment (makelicious) [inline src/form/v2/birth/advancedSearch.ts:130]: Could we use actual year rather than 365 days

File: src/form/v2/birth/advancedSearch.ts (hunk 2 above)
Comment (tareq89) [reply src/form/v2/birth/advancedSearch.ts:106]: EventMetadataSearchOptions are moved to a single file since they will be common for all types of events.

File: src/form/v2/birth/advancedSearch.ts (hunk 3 above)
Comment (tareq89) [reply src/form/v2/birth/advancedSearch.ts:130]: yes, for month, quarters we have used date library function too!
